src/joint_state_listener.cpp

```diff
@@ -171,8 +171,12 @@ int main(int argc, char** argv)
 
   // gets the location of the robot description on the parameter server
   urdf::Model model;
-  if (!model.initParam("robot_description"))
+  if (!model.initParam("robot_description")) {
+    if (model.links_.empty()) {
```

Comment:
This seems to be in a weird location; we are only checking if the links are empty if we *failed* to initialize the URDF.  But in that case, we are going to quit anyway since we can't do anything.  Doesn't this check want to be down below, after a successful `initParam`?

Reply:
Hi @clalancette and thank you for the super quick review :smile: 
I'll explain: [initParam](https://github.com/ros/urdf/blob/470729633ba0419735603a054fec0b5bbfe442f9/urdf/src/model.cpp#L85) will fail on urdf without any links because it calles [initString](https://github.com/ros/urdf/blob/470729633ba0419735603a054fec0b5bbfe442f9/urdf/src/model.cpp#L119) which in turn call [parseURDF](https://github.com/ros/urdf/blob/470729633ba0419735603a054fec0b5bbfe442f9/urdf/src/model.cpp#L201)... [which will fail and print an error if no links](https://github.com/ros/urdfdom/blob/06f5f9bc34f09b530d9f3743cb0516934625da54/urdf_parser/src/model.cpp#L178). so adding the check after it will have no effet.

Reply:
Hm, OK.  I guess I'm a bit confused about this PR then.  Before this PR, the code looked like this:

```
  if (!model.initParam("robot_description"))
    return -1;
```

Thus, if `parseURDF` returned a nullptr, which caused `initParam` to return false, wouldn't we *already* be avoiding a segmentation fault?  That is, what is the difference that this PR make?

Reply:
Hm, so the reason I'm so confused is that in the original report, @sloretz was seeing a segmentation fault (that's the -11 error code he was originally reporting).  But with the current robot_state_publisher (without this PR at all), I can't reproduce that; it just exits with:

```
[ERROR] [1547660920.315867675]: No link elements found in urdf file
[robot_state_publisher-2] process has died [pid 12998, exit code 255, cmd /home/ubuntu/melodic/catkin_ws/install_isolated/lib/robot_state_publisher/robot_state_publisher __name:=robot_state_publisher __log:=/home/ubuntu/.ros/log/f4c98e34-19b6-11e9-9480-00163e5a3681/robot_state_publisher-2.log].
```

So now I'm wondering whether we need this PR (and this issue) at all; we are *already* checking that it is valid, and exiting.  The only thing this PR will change is the return code, which may be worthwhile, but is a lot smaller scope than the original.

@sloretz Any thoughts here?

Reply:
It looks like the ticket I filed has a path to `indigo`. Maybe I was using an outdated version of robot state publisher?

Reply:
@sloretz @clalancette yes, looks like it was fixed in https://github.com/ros/robot_state_publisher/commit/b9029f7d5830aceeabfdd94048fb4df4def2d9e0

Reply:
@betab0t thanks for the PR and discovering that commit! I'll close the original issue, sorry for the confusion.

Reply:
@betab0t That sounds great, it is much appreciated.